file/pkg/service/service.go

```diff
@@ -80,3 +82,44 @@ func (f *fileService) ListFileShare(ctx context.Context, in *pb.ListFileShareReq
 	log.Debugf("Listed File Share successfully, #num=%d, fileshares: %+v\n", len(fileshares), fileshares)
 	return nil
 }
+
+func (f *fileService) GetFileShare(ctx context.Context, in *pb.GetFileShareRequest, out *pb.GetFileShareResponse) error {
+	log.Info("Received GetFileShare request.")
+
+	fs, err := db.DbAdapter.GetFileShare(ctx, in.Id)
+	if err != nil {
+		log.Errorf("failed to get fileshare: %v\n", err)
+		return err
+	}
+	var tags []*pb.Tag
+	for _, tag := range fs.Tags {
+		tags = append(tags, &pb.Tag{
+			Key:   tag.Key,
+			Value: tag.Value,
+		})
+	}
+	out.Fileshare = &pb.FileShare{
+		Id:                 fs.Id.Hex(),
+		CreatedAt:          fs.CreatedAt,
+		UpdatedAt:          fs.UpdatedAt,
+		Name:               fs.Name,
+		Description:        fs.Description,
+		TenantId:           fs.TenantId,
+		UserId:             fs.UserId,
+		BackendId:          fs.BackendId,
+		Backend:            fs.Backend,
+		Size:               fs.Size / utils.GB_FACTOR,
```

Comment:
Why the conversion here? Let's report in bytes and leave it to client to convert

Reply:
Yes, The API takes Size in GiB and We are persisting in our DB in bytes.
So the conversion is required.

In case of Azure, The API send size in GiB. So keeping the request response consistent. Conversion is required for reporting it back as requested in the request body.

Reply:
Got it..But better we report it in Bytes and let the API consumer convert it into MB, GB,TB or etc.. It will make consistent. thanks

Reply:
Code changes Done.
Thanks